src/MinimalDelegation.sol

```diff
@@ -44,16 +45,15 @@ contract MinimalDelegation is IERC7821, IKeyManagement, ERC1271, EIP712, ERC4337
     }
 
     /// @dev The mode is passed to allow other modes to specify different types of opData decoding.
-    function _authorizeOpData(bytes32, Call[] calldata calls, bytes calldata opData) private view {
+    function _authorizeOpData(bytes32, Call[] calldata calls, bytes calldata opData) private {
         // TODO: Can switch on mode to handle different types of authorization, or decoding of opData.
         // For now, we only support decoding necessary information needed to verify 1271 signatures.
-        (, bytes calldata signature) = opData.decodeUint256Bytes();
-        // TODO: Nonce validation.
+        (uint256 nonce, bytes calldata signature) = opData.decodeUint256Bytes();
+        validateAndUpdateNonce(nonce);
         // Check signature.
-        bool isValid;
-        /// The calls are not safe hashed with _hashTypedData, as the domain separator is sufficient to prevent against replay attacks.
-        (isValid,) = _isValidSignature(calls.hash(), signature);
+        (bool isValid,) = _isValidSignature(_hashTypedData(calls.hash(nonce)), signature);
         if (!isValid) revert IERC7821.InvalidSignature();
+        // TODO: Store keyHash in storage.
```

Comment:
what is this TODO for?

Reply:
oh weird, i thought that was already there - don't know how that got there